pkg/controller/plan/validation.go

```diff
@@ -814,6 +823,33 @@ func (r *Reconciler) validateVM(plan *api.Plan) error {
 			invalidDiskSizes.Items = append(invalidDiskSizes.Items, ref.String())
 		}
 
+		conflicts, err := validator.MacConflicts(*ref)
+		if err != nil {
+			return err
+		}
+		if len(conflicts) > 0 {
+			macConflicts.Items = append(macConflicts.Items, ref.String())
+			// Group conflicts by destination VM for this specific source VM
+			vmConflictsByVM := make(map[string][]string)
+			for _, conflict := range conflicts {
+				vmConflictsByVM[conflict.DestinationVM] = append(vmConflictsByVM[conflict.DestinationVM], conflict.MAC)
+			}
+
+			// Build detailed message with grouped conflicts
+			var conflictDetails []string
+			for destinationVM, macs := range vmConflictsByVM {
+				if len(macs) == 1 {
+					conflictDetails = append(conflictDetails, fmt.Sprintf("MAC %s conflicts with destination VM %s", macs[0], destinationVM))
+				} else {
+					conflictDetails = append(conflictDetails, fmt.Sprintf("MACs %s conflict with destination VM %s", strings.Join(macs, ", "), destinationVM))
+				}
```

Comment:
Don't think you need this if case, as the `strings.Join` wiht just one item joins without appliing the `", "`